src/actions/layers/Layers.ts

```diff
@@ -2,29 +2,65 @@ import {ILayerAction} from "./ILayerAction";
 import Action from "../Action";
 // TODO - BundleSize Warning - we include all the Sources code within Layers.
 import Source, {ImageSource} from "./Source";
+import {Position} from "../../params/position/Position";
+import Param from "../../parameters/Param";
 
 
 class Layer extends Action implements ILayerAction {
   source: ImageSource; // TODO this needs to accept other types of sources
-  constructor(transformable: ImageSource) {
+  position:Position;
+  constructor(transformable: ImageSource, position:Position) {
     super();
     this.source = transformable;
+    this.position = position;
+  }
+
+  /**
+   * Layers are built using three bits -> /Open/Transform/Close
+   * The opening of a layer
+   */
+  openLayer() {
+    return `l_${this.source.asset.publicID}`;
+  }
+
+  /**
+   * Layers are built using three bits -> /Open/Transform/Close
+   * Transformations conducted on the image in the layer
+   */
+  layerTransformation() {
+    return this.source.toString();
+  }
+
+  /**
+   * Layers are built using three bits -> /Open/Transform/Close
+   * Closing of the layer, includes Position as well
+   */
+  closeLayer() {
+    const bit = new Action().addParam(new Param('fl', 'layer_apply'));
```

Comment:
Some actions use layer and utilize falgs (Cutter action is using fl_cutter) - are the flags handled in the `addParam` function? otherwise you need to make sure the flags are concatenated (`fl_cutter.layer_apply`).

Reply:
If these flags are all added on the layer action, this shouldn't be a problem.

Reply:
The flags are part of the layer action itself, it should be self-contained.
If the user wants to add a flag to the resulting action at some later point that's a a different story and unrelated anyway (and also pretty much undefined when it comes to layers...).

LGTM